src/main/java/nextstep/security/filter/BasicAuthenticationFilter.java

```diff
@@ -0,0 +1,45 @@
+package nextstep.security.filter;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.security.authentication.*;
+import nextstep.security.user.UserDetailsService;
+import nextstep.security.util.Base64Convertor;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.List;
+
+public class BasicAuthenticationFilter extends OncePerRequestFilter {
+    private final AuthenticationManager authenticationManager;
+
+    public BasicAuthenticationFilter(UserDetailsService userDetailsService) {
+        this.authenticationManager = new ProviderManager(List.of(new DaoAuthenticationProvider(userDetailsService)));
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        try {
+            String authorization = request.getHeader("Authorization");
```

Comment:
`HttpHeaders.AUTHORIZATION`  를 사용해봐도 좋을 것 같아요 😄 
```suggestion
            String authorization = request.getHeader(HttpHeaders.AUTHORIZATION);
```